services/galley/src/Galley/Aws.hs

```diff
@@ -155,11 +157,13 @@ execute e m = liftIO $ runResourceT (runReaderT (unAmazon m) e)
 enqueue :: E.TeamEvent -> Amazon ()
 enqueue e = do
     QueueUrl url <- view eventQueue
-    res <- retrying (limitRetries 5 <> exponentialBackoff 1000000) (const canRetry) $ const (sendCatch (req url))
+    rnd <- liftIO nextRandom
+    res <- retrying (limitRetries 5 <> exponentialBackoff 1000000) (const canRetry) $ const (sendCatch (req url rnd))
     either (throwM . GeneralError) (const (return ())) res
   where
     event = decodeLatin1 $ B64.encode $ encodeMessage e
-    req url = SQS.sendMessage url event & SQS.smMessageGroupId .~ Just "team.events"
+    req url dedup = SQS.sendMessage url event & SQS.smMessageGroupId .~ Just "team.events"
```

Comment:
Some of the SQS tests were flaky due to the de-duplication of messages on SQS; since our messages often look the same (e.g., `TEAM_CREATE` with the same `team id` will always have the same timestamp) so it's useful to add a unique id (which is reused in case of queueing failures _of the same message_)